engine/src/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUT.java

```diff
@@ -722,34 +700,29 @@ public Result execute( Result previousResult, int nr ) {
         }
       }
 
-      // Get all the files in the local directory...
-      int x = 0;
-
-      // Joerg: ..that's for Java5
-      // ArrayList<String> myFileList = new ArrayList<String>();
-      ArrayList<String> myFileList = new ArrayList<String>();
+      // handle file:/// prefix
+      if ( realLocalDirectory.startsWith( "file:" ) ) {
```

Comment:
Seems like this could generate a NPE if nobody set the realLocalDirectory? If so please add unit tests for this being null and not null. Same goes for the other variables filled with environmentSubstitute() above, thanks!